### PR TITLE
Let `textual run` run files that don't end in .py

### PR DESCRIPTION
**Pull request by davep**
It's not uncommon to create "standalone" commands in Python that don't end in a .py and which:

  `#!/usr/bin/env python`

or a variation on that theme. This commit makes it so that `textual run` will treat a file that's named and set to run like that as if it were a file that ends in .py.

See #1106.

### CODE REVIEW

File: src/textual/_import_app.py [hunk 1]
```diff
@@ -16,6 +16,23 @@ class AppFail(Exception):
     pass
 
 
+def shebang_python(candidate: Path):
```
Comment (willmcgugan) [inline src/textual/_import_app.py:19]: Return type por favor.

File: src/textual/_import_app.py [hunk 2]
```diff
@@ -16,6 +16,23 @@ class AppFail(Exception):
...
+    Args:
+        candidate (Path): The candidate file to check.
+
+    Returns:
+        bool: ``True`` if it looks to #! python, ``False`` if not.
+    """
+    try:
+        with candidate.open("rb") as source:
+            first_line = source.readline()
+        return first_line.startswith(b"#!") and b"python" in first_line
```
Comment (willmcgugan) [inline src/textual/_import_app.py:31]: Could you put the return after the except? Just to keep the exception block as focused as possible.

**Review by willmcgugan — CHANGES_REQUESTED** [review]
small requests.

File: src/textual/_import_app.py (hunk 1 above)
Comment (davep) [reply src/textual/_import_app.py:19]: 🤦

File: src/textual/_import_app.py (hunk 2 above)
Comment (davep) [reply src/textual/_import_app.py:31]: Aye, good call, will do.

File: src/textual/_import_app.py (hunk 1 above)
Comment (davep) [reply src/textual/_import_app.py]: Done.

File: src/textual/_import_app.py (hunk 2 above)
Comment (davep) [reply src/textual/_import_app.py]: Done.

**Pull request merged**